src/apps/vpn/controller.cpp

```diff
@@ -186,14 +189,20 @@ void Controller::implInitialized(bool status, bool a_connected,
   }
 }
 
-bool Controller::activate() {
+bool Controller::activate(const ServerData& serverData) {
   logger.debug() << "Activation" << m_state;
 
   if (m_state != StateOff && m_state != StateSwitching) {
     logger.debug() << "Already connected";
     return false;
   }
 
+  m_serverData = serverData;
```

Comment:
I'm finding it a little confusing about what is m_nextServerData and what is m_serverData.

Further down this file when we look at `switchServers`. I think the following happens:
* we request switchServers
* we set the property m_nextServerData
* we setState to `StateSwitching`
* we run deactivate, this leads to triggering of disconnected
* in disconnected we look to see if we're in StateSwitching and if yes we call activate with m_nextServerData which is what finally makes next the current server data.

1 - i'm finding it hard to reason about this state machine. maybe it's just me. i was looking for documentation like https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/docs/authentication-in-app.md#the-whole-flow - does it exist?
2 - here in activate we set m_serverData but we haven't actually successfully done the activation yet. I'm feeling like I want a Controller::connected() slot that sets m_serverData after a successful activation

Can you help me with my understanding / mental model?

Reply:
Yes, I wrote a better (I hope so at least!) comment to describe the logic. 
To summarize it here:
- m_serverData is a copy of the location, taken when the user (or something else) triggers a VPN activation. This value is kept constant for the whole duration of the operation. If something happens, this and MozillaPVN::serverData are in sync.
- If there is a server-switch request while an activation is in progress, we start a disconnect operation and we store the new location in m_nextServerData.

See this flow:

```mermaid
sequenceDiagram
    User->>Client: Location X is set
    User->>Client: VPN activation request
    activate Client
    Client->>Client: Task scheduled (copy of location is taken: Xc = X)
    deactivate Client
    Client->>Controller: Task run: activate (Xc)
    activate Controller
    Controller->>Controller: m_serverData = Xc
    Controller->>Controller: Connecting...
    User->>Client: Location Y is set
    activate Client
    Client->>Client: Task scheduled (copy of location is taken: Yc = Y)
    deactivate Client
    Controller->>Controller: Conferming...
    Client->>Controller: Task run: serverSwitch (Yc)
    Controller->>Controller: m_nextServerData = Yc
    Controller->>Controller: Connected
    Controller->>Controller: deactivate...
    Controller->>Controller: ativate(m_nextServerData)
    Controller->>Controller: m_serverData = m_nextServerData
    deactivate Controller
```

Reply:
Given that you've made this awesome diagram. Should it go in the docs folder?